.github/workflows/self-push.yml

```diff
@@ -89,68 +59,38 @@ jobs:
           name: run_all_tests_torch_gpu_test_reports
           path: reports
 
-
   run_tests_tf_gpu:
-    runs-on: [self-hosted, gpu, single-gpu]
+    runs-on: [self-hosted, docker-gpu, single-gpu]
+    container:
+      image: tensorflow/tensorflow:2.4.1-gpu
+      options: --gpus all --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
```

Comment:
Do you need `--gpus all` if you want to run only on a single GPU? 

You can do `--gpus  0` and then remove relying on `CUDA_VISIBLE_DEVICES: 0`

Reply:
We must make it 1 gpu or as you said set `CUDA_VISIBLE_DEVICES: 0` because some tests will be skipped if it finds multiple-gpus.

Reply:
Yes, it's currently using `CUDA_VISIBLE_DEVICES: 0` but that's just a check as these runs only run on single-GPU machines anyway. I can add `--gpus 0` instead of `CUDA_VISIBLE_DEVICES: 0`, but again these two should be unnecessary as the tag `single-gpu` prevents these from being run on multi-GPU machines.

Reply:
If it's not too much work, my vote is to make it explicit, since the snippets later get copied and when reading several of these in parallel it's easier to quickly grasp that this is 1-gpu recipe.